### PR TITLE
[SYCL] Forbid non-const static variable uses in device code

### PR DESCRIPTION
**Pull request by fadeeval**
Adding the error, that forbid use of non-const static variable inside kernel.

Signed-off-by: Aleksander Fadeev aleksander.fadeev@intel.com

### CODE REVIEW

**Comment by fadeeval** [comment]
> I'm OK with it.
> Take a look to the clang-format failures.

Okey.

File: clang/lib/Sema/SemaExpr.cpp [hunk 1]
```diff
@@ -210,6 +210,15 @@ bool Sema::DiagnoseUseOfDecl(NamedDecl *D, ArrayRef<SourceLocation> Locs,
...
                              ObjCInterfaceDecl *ClassReceiver) {
+
+  if (getLangOpts().SYCLIsDevice) {
+    if (auto VD = dyn_cast<VarDecl>(D)) {
+      if (VD->getStorageClass() == SC_Static &&
+          !VD->getType().isConstant(Context))
+        SYCLDiagIfDeviceCode(*Locs.begin(), diag::err_sycl_restrict)
+            << Sema::KernelNonConstStaticDataVariable;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (fadeeval) [reply clang/lib/Sema/SemaExpr.cpp]: Ok

[the remark this answers is not in the document]
Comment (fadeeval) [reply clang/lib/Sema/SemaExpr.cpp]: Should I delete all curve braces?

[the remark this answers is not in the document]
Comment (bader) [reply clang/lib/Sema/SemaExpr.cpp]: I proposed just move empty line from "before `if`" to "after `if`". That's it.

[the remark this answers is not in the document]
Comment (fadeeval) [reply clang/lib/Sema/SemaExpr.cpp]: Got it.

**Pull request merged**